Assets/LeapMotionModules/ElementRenderer/Scripts/Utility/PackUtil.cs

```diff
@@ -96,31 +97,26 @@ public class Settings {
     for (int i = 0; i < textureArray.Length; i++) {
       if (textureArray[i] == null) {
         textureArray[i] = defaultTexture;
-      } else {
-        textureArray[i].EnsureReadWriteEnabled();
       }
     }
 
     packedTexture = new Texture2D(1, 1, format, mipmap: false);
     packedTexture.filterMode = settings.filterMode;
   }
 
-  private static Dictionary<BorderKey, Texture2D> _cachedBorderedTextures = new Dictionary<BorderKey, Texture2D>();
-  private static Texture2D getBordered(Texture2D source, int border) {
-    if (border <= 0) {
-      return source;
-    }
-
+  private static Dictionary<ProcessKey, Texture2D> _cachedProcessedTextures = new Dictionary<ProcessKey, Texture2D>();
+  private static Texture2D processTexture(Texture2D source, int border) {
     if (source == null) {
       return null;
     }
 
     Texture2D bordered;
-    BorderKey key = new BorderKey() { texture = source, border = border };
-    if (!_cachedBorderedTextures.TryGetValue(key, out bordered)) {
-      source.EnsureReadWriteEnabled();
-      bordered = source.GetBordered(border);
-      _cachedBorderedTextures[key] = bordered;
+    ProcessKey key = new ProcessKey() { texture = source, border = border };
+    if (!_cachedProcessedTextures.TryGetValue(key, out bordered)) {
+      var tempReadable = source.GetReadableTexture();
```

Comment:
Makes sense, I'll add in a clause that allows GetReadableTexture to return the texture itself if it is readable.  As for the ARGB32 issue, currently that is a known issue (#563), because Texture2D.Pack changes the texture format to argb32 no matter what the input textures are :(